plugin/filetype.vim

```diff
@@ -1,5 +1,5 @@
 augroup filetypedetect
 
-au BufNewFile,BufRead * lua require('filetype').resolve()
+au BufNewFile,BufRead *.* lua require('filetype').resolve()
```

Comment:
This change is unrelated to the subject of this PR, it should probably be extracted to it’s own. Also is it necessary? It seems to me this is restricting the set of matches to files that have a `.` in their name, but not all files that have associated file types have a dot, for example `/etc/fstab`.

Reply:
Agreed. I think I'll just rename it to `filename_mappings.lua`